### DISCUSSION TITLE
Mina snapp verification question

### DISCUSSION THREAD

**Discussion by frisitano** [NilFoundation/evm-mina-verification#38] [Q&A]
Hi,

Very excited to see this great work around evm verification of Mina state come to life.  I was wondering if this framework could be applied to evm verification of a Mina snapp (mina smart contract) directly without the proof having to pass through the mina blockchain and into mina state?  The reason I would like this is because it could be a way to achieve low latency verification of snapps in evm.  Even if this is not necessarily possible with this framework do you know if it is theoretically possible to build something of this nature?

Many thanks.

**Comment by nemothenoone** [comment]
Hey, @frisitano!

What you're talking about is theoretically possible, but it makes no sense. Why?

Well, a circuit generated out of Mina-specific definition (they call it snapp) would in best case result into the simple Pickles proof, which proves some data change. But! Since it is not attached to Mina's state, the proof itself would make no sense, because it would be only about proving data change, but without the initial data state. So such a proof verification would be unrelated to Mina at all.

There is a way to put all the necessary data directly to Ethereum, but in this case the whole thing would simply result into something like StarkNet turning out to be unrelated to Mina again.

**Comment by frisitano** [comment]
Hey @nemothenoone,

thanks for getting back to me and thank you for clarifying!  All the best with the development :)
